### ISSUE TITLE
False positive ``used-before-assignment`` on different line returns 

### ISSUE THREAD

**Issue by otchita** [PyCQA/pylint#4301]
This is related to my previous issue #4238 for the support of the walrus operator. Another edge case seems to be unhandled. That of different line returns.

### Steps to reproduce

Given a file `good-test.py`:
```python
"""Good test example."""
def func():
    """Test."""
    return f'The number {(count := 4)} is equal to {count}'

```
`pylint` doesn't detect any errors which is expected. However when return to a line like this:

Given a file `bad-test.py`:
```python
"""Bad test example."""
def func():
    """Test."""
    return f'The number {(count := 4)} ' \
           f'is equal to {count}'

```
We obtain the following:
### Current behavior

Result of `pylint bad-test.py`:
```
************* Module test
bad-test.py:4:19: E0601: Using variable 'count' before assignment (used-before-assignment)
```

### Expected behavior

Normally, this should not raise any errors.

### pylint --version output

Result of pylint --version output on my local macOS:

```
pylint 2.7.4
astroid 2.5.2
Python 3.8.8 (v3.8.8:024d8058b0, Feb 19 2021, 08:48:17) 
[Clang 6.0 (clang-600.0.57)]
```

**Comment by cdce8p** [comment]
Thanks for the report! The issue will be fixed with  #4307
